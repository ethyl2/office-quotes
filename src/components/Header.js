import React from 'react';
import logo from '../logo.svg';

class Header extends React.Component {
    render() {
        return (
            <header className="App-header">
                <img src={logo} className="App-logo" alt="logo" />
                <h1>Quotes from The Office</h1>  
            </header>
        )
    }
}

export default Header;