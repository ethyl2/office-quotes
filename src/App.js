import React, { useState } from 'react';

import Header from './components/Header';

import './App.css';

function App() {
  const [message, setMessage] = useState('');
  
  const fakeApiCall = () => Promise.resolve('Success!');

  const onClickHandler = () => {
    fakeApiCall().then(res => setMessage(res));
  }

  return (
    <div className="App">
      <Header />
      <div>
        <span>{message}</span>
        <button onClick={onClickHandler}>Get Message!</button>
      </div>
    </div>
  );
}

export default App;
