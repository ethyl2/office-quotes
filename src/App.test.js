import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Header from './components/Header';
import { act } from 'react-dom/test-utils';

/*
test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
})


describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
*/

describe('<Header />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});

test('<App /> matches snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders success text', () => {
  const { getByText, findByText } = render(<App />);
  act(() => {
    fireEvent.click(getByText('Get Message!')); 
  });
  findByText(/success/i);
  //const tester = getByText(/success/i);
  //expect(tester).toBeInTheDocument();
  //console.log(successEl);
});

