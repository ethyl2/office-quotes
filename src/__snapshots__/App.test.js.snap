// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1>
      Experimenting with tests.
    </h1>
    <h2>
      Including snapshot testing, using react-test-renderer
    </h2>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <div>
    <span>
      
    </span>
    <button
      onClick={[Function]}
    >
      Get Message!
    </button>
  </div>
</div>
`;

exports[`<Header /> should match snapshot 1`] = `
<header
  className="App-header"
>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <h1>
    Experimenting with tests.
  </h1>
  <h2>
    Including snapshot testing, using react-test-renderer
  </h2>
  <a
    className="App-link"
    href="https://reactjs.org"
    rel="noopener noreferrer"
    target="_blank"
  >
    Learn React
  </a>
</header>
`;
